index.bs

```diff
@@ -405,38 +400,66 @@ spec: webidl
     };
   </pre>
   <p>
-    The steps to <dfn>retrieve a permission storage entry</dfn> of a
-    <a>permission storage identifier</a> are as follows:
+    The steps to <dfn export>retrieve the permission storage</dfn> of a
+    {{PermissionName}} <var>name</var> are as follows:
   </p>
   <ol>
-    <li>If the <a>user agent</a> has a {{PermissionStorage}} associated
-    with the <a>permission storage identifier</a> in its permission store,
-    it MUST return the {{PermissionStorage}}.
+    <li>Let <var>settings</var> be the <a>current settings object</a>.</li>
+    <li>
+      Let <var>store</var> be some <a>permission store</a> associated with the
+      <a>same origin</a> as <var>settings</var>. This store SHOULD be chosen
+      consistently for a given <a>environment settings object</a>.
+    </li>
+    <li>
+      If <var>store</var> exists and maps <var>name</var> to a value, return
+      this value.
     </li>
-    <li>Otherwise, it MUST return <code>undefined</code>.
+    <li>
+      Otherwise, return a default value based on the <a>user agent</a>'s defined
+      heuristics. For example, <code>{state: {{"prompt"}}}</code> can be a
+      default value, but it can also be based on frequency of visits.
     </li>
   </ol>
   <p>
-    The steps to <dfn>create a permission storage entry</dfn> for a
-    <a>permission storage identifier</a> are as follows:
+    The steps to <dfn>create a permission storage entry</dfn> mapping a
+    {{PermissionName}} <var>name</var> to a {{PermissionStorage}}
+    <var>storage</var> are as follows:
   </p>
   <ol>
-    <li>If the <a>user agent</a> has a {{PermissionStorage}} associated
-    with the <a>permission storage identifier</a> in its permission store,
-    it MUST overwrite it to the given {{PermissionStorage}}.
+    <li>Let <var>settings</var> be the <a>current settings object</a>.</li>
+    <li>
+      Let <var>stores</var> be a collection of zero or more <a>permission
+      store</a>s associated with the <a>same origin</a> as <var>settings</var>.
+      If <var>stores</var> is non-empty, it SHOULD include the permission store
+      used in the <a>retrieve the permission storage</a> algorithm for
+      <var>settings</var>.
     </li>
-    <li>Otherwise, it MUST write the new {{PermissionStorage}} to its
-    permission store.
+    <li>
+      Write a mapping from <var>name</var> to <var>storage</var> into each
+      <a>permission store</a> in <var>stores</var>, overwriting any existing
+      mapping with the same <var>name</var>.
+
+      Issue(91): This isn't quite sufficient for complex {{PermissionStorage}}
+      subclasses, which may need to modify the old storage instead of just
+      overwriting it.
```

Comment:
Oh, I meant "Let stores be a collection of zero or more permission stores associated with the same origin as settings." to allow the UA to pick whichever subset it wants, possibly differently for each operation.